src/Blog/Entity/Post.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Blog\Entity;
+
+class Post
+{
+    /**
+     * @var int
+     */
+    public $id;
+
+    /**
+     * @var string
+     */
+    public $title;
+
+    /**
+     * @var text
+     */
+    public $content;
+
+    /**
+     * @var string
+     */
+    public $picture;
+
+    /**
+     * @var enum
+     * @options=["draft","published"]
+     */
+    public $status;
+
+    /**
+     * @var datetime
+     */
+    public $created_at;
+
+    /**
+     * @var datetime
+     */
+    public $updated_at;
+
+    /**
+     * @var int
+     */
+    public $userId;
+
+    public function __construct()
+    {
+        $this->created_at = new \Datetime($this->created_at);
```

Comment:
à ce moment, `$this->created_at` est null car tu es dans le constructeur

si tu veux mettre la date et heure du moment, tu peux faire : `new \Datetime('now')`

Reply:
nan ce n'est pas pour définir la date du moment, mais pour definir comme datetime le created_at de la base de données!
J utilise ces données avec la fonction ago (en js), si je ne fais pas cela, il me retourne une date au format string et non datetime.
il faut mieux que j'utilise un setter?